lib/saml_idp/name_id_formatter.rb

```diff
@@ -15,15 +18,21 @@ def all
     end
 
     def chosen
-      if split?
-        version, choose = "1.1", one_one.first
-        version, choose = "2.0", two_zero.first unless choose
-        version, choose = "2.0", "persistent" unless choose
-        build(version, choose)
-      else
-        choose = list.first || "persistent"
-        build("2.0", choose)
-      end
+      symbolized_name_id_format = sp_name_id_format.split(':').last.underscore.to_sym
+
+      return default_name_getter_hash unless list.key?(symbolized_name_id_format)
+
+      version = symbolized_name_id_format == :email_address ? '1.1' : '2.0'
```

Comment:
There are a few other nameid-format URIs that have `1.1` as the version:

- `urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified`
- `urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress`
- `urn:oasis:names:tc:SAML:1.1:nameid-format:X509SubjectName`
- `urn:oasis:names:tc:SAML:1.1:nameid-format:WindowsDomainQualifiedName`

Section 8.3 Name Identifier Format Identifiers:
https://www.oasis-open.org/committees/download.php/56776/sstc-saml-core-errata-2.0-wd-07.pdf

The code in `#build` will also incorrectly lowercase `X509SubjectName` and `Windows...`, not sure that's worth fixing.

Reply:
Fixed here: https://github.com/18F/saml_idp/pull/10/commits/6b99bd85975e6770e887657c62389d45b573408e